### PR TITLE
Add reuseable actions

### PR DESCRIPTION
**Pull request by MSevey**
# PULL REQUEST

## Overview
Add composite actions that can be used in other repos.

### CODE REVIEW

File: actions/js-lint/action.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+# Define action name, description, and inputs
+name: "JS Lint"
+description: "Lint code with Prettier and eslint"
+inputs:
+  working-directory:
+    required: false
+    description: "Working directory"
+    default: "."
```
Comment (kwypchlo) [inline actions/js-lint/action.yml:8]: I think working-directory is a standard github action input and we don't need to specify it, do we ?

File: actions/python-black-lint/action.yml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  steps:
+    - uses: actions/setup-python@v2
+      with:
+        python-version: "3.x"
+        architecture: x64
+
+    - run: pip install black
+      shell: bash
+    - run: black --check .
+      shell: bash
```
Comment (kwypchlo) [inline actions/python-black-lint/action.yml:17]: I don't think you need to defined shell specifically here

```suggestion
    - run: pip install black
    - run: black --check .
```

File: actions/yamllint/action.yml [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+  steps:
+    - uses: actions/setup-python@v2
+      with:
+        python-version: "3.x"
+        architecture: x64
+
+    - run: pip install yamllint
+      shell: bash
+    - run: yamllint --no-warnings .
+      shell: bash
```
Comment (kwypchlo) [inline actions/yamllint/action.yml:17]: I don't think you need to defined shell specifically here

```suggestion
    - run: pip install yamllint
    - run: yamllint --no-warnings .
```

File: workflows/dockerfile-lint.yml [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+        type: string
+        description: "path to dockerfile"
+        default: "Dockerfile"
+
+jobs:
+  hadolint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: hadolint/hadolint-action@v2.0.0
```
Comment (kwypchlo) [inline workflows/dockerfile-lint.yml:17]:
```suggestion
      - uses: hadolint/hadolint-action@v2.1.0
```

File: actions/version-release/action.yml [hunk 5]
```diff
@@ -0,0 +1,27 @@
+name: "Version Release"
```
Comment (kwypchlo) [inline actions/version-release/action.yml:1]: What will be the trigger to call this action? Not sure about the use case.

File: actions/python-flake8-lint/action.yml [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+
+    - run: pip install flake8
+      shell: bash
+    # E203: https://www.flake8rules.com/rules/E203.html - Whitespace before ':'
+    # E501: https://www.flake8rules.com/rules/E501.html - Line too long
+    # W503: https://www.flake8rules.com/rules/W503.html - Line break occurred before a binary operator
+    # W605: https://www.flake8rules.com/rules/W605.html - Invalid escape sequence
+    # E722: https://www.flake8rules.com/rules/E722.html - Do not use bare except, specify exception instead
+    - run: flake8 --max-line-length 88 --ignore E203,E501,W503,W605,E722
+      shell: bash
```
Comment (kwypchlo) [inline actions/python-flake8-lint/action.yml:22]: I don't think you need to defined shell specifically here

```suggestion
```

File: actions/python-flake8-lint/action.yml [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+runs:
+  using: "composite"
+  steps:
+    - uses: actions/setup-python@v2
+      with:
+        python-version: "3.x"
+        architecture: x64
+
+    - run: pip install flake8
+      shell: bash
```
Comment (kwypchlo) [inline actions/python-flake8-lint/action.yml:15]: I don't think you need to defined shell specifically here

```suggestion
```

**Review by kwypchlo — CHANGES_REQUESTED** [review]

File: actions/python-black-lint/action.yml (hunk 2 above)
Comment (MSevey) [reply actions/python-black-lint/action.yml:17]: You do need to specific it with composite actions or you will get an error.

File: actions/version-release/action.yml (hunk 5 above)
Comment (MSevey) [reply actions/version-release/action.yml:1]: Here is an example of how I plan to use this action

https://github.com/SkynetLabs/ansible-playbooks/pull/286/files#diff-87db21a973eed4fef5f32b267aa60fcee5cbdf03c67fafdc2a9b553bb0b15f34R47

This is part of enable continuous release for our repos

**Review by firyx — APPROVED** [review]

**Review by kwypchlo — APPROVED** [review]

**Pull request merged**